File: app/src/main/java/com/developer/fabian/feedrss/ui/adapter/MainListAdapter.java
package com.developer.fabian.feedrss.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

import java.util.ArrayList;

public abstract class MainListAdapter extends BaseAdapter {

    private ArrayList<?> enterList;
    private int idView;
    private Context context;

    protected MainListAdapter(ArrayList<?> enterList, int idView, Context context) {
        super();

        this.enterList = enterList;
        this.idView = idView;
        this.context = context;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(idView, null);
        }

        onEnter(enterList.get(position), convertView);

        return convertView;
    }

    @Override
    public int getCount() {
        return enterList.size();
    }

    @Override
    public Object getItem(int position) {
        return enterList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    public abstract void onEnter(Object enter, View view);
}

File: app/src/main/java/com/developer/fabian/feedrss/ui/adapter/MainListAdapter.kt
package com.developer.fabian.feedrss.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter

import java.util.ArrayList

abstract class MainListAdapter protected constructor(private val enterList: ArrayList<*>,
                                                     private val idView: Int,
                                                     private val context: Context) : BaseAdapter() {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView

        if (view == null) {
            val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(idView, null)
        }

        onEnter(enterList[position], view!!)

        return view
    }

    override fun getCount(): Int {
        return enterList.size
    }

    override fun getItem(position: Int): Any {
        return enterList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    abstract fun onEnter(enter: Any, view: View)
}
